### PR TITLE
fix(helm):  Display debug release info when upgrade failed because of timeout. Closes #3020

### LINKED ISSUE #3020: Using --wait suppresses debug logs
This makes it very hard to investigate what is going on with a bad helm upgrade

### PR DESCRIPTION
**Pull request by splisson**
When an upgrade fails because of a timeout waiting for it to be ready, in debug mode, output the release content.

Closes #3020

### CODE REVIEW

**Review by thomastaylor312 — APPROVED** [review]
This looks good to me. Sorry for the really long delay, this somehow slipped through the cracks in my notifications.

**Comment by thomastaylor312** [comment]
@bacongobbler Before I merge, do you think this is something we could bring into the next bug fix release?

**Comment by splisson** [comment]
No problem. Thanks for your work maintaining helm

Sébastien

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
@splisson Hey, sorry to reset my review on this, but can you add this same logic to the `install` and `rollback` commands? It works great in `upgrade`, but it should have parity across the other commands as well

File: cmd/helm/install.go [hunk 1]
```diff
@@ -264,23 +264,33 @@ func (i *installCmd) run() error {
 		helm.InstallDisableHooks(i.disableHooks),
 		helm.InstallTimeout(i.timeout),
 		helm.InstallWait(i.wait))
-	if err != nil {
-		return prettyError(err)
+
+	// If there is an error while waiting, make a call without waiting to get the release content
+	if (resp == nil || resp.Release == nil) && i.wait {
```
Comment (thomastaylor312) [inline cmd/helm/install.go]: Should this be wrapped with `settings.Debug`?

Comment (thomastaylor312) [reply cmd/helm/install.go]: Actually, I looked some more into this. I think it should be `i.printRelease` which is where it prints debug

**Comment by thomastaylor312** [comment]
Ok @splisson, now that I have my head on straight and am not trying to do multiple things at once, it looks like `rollback` never printed anything before and I thought it did. Also, the install needs to use `i.printRelease`, which isn't intuitive I know. So I would remove the change to rollback and make a slight adjustment to the `install` logic and then this is good to go. 

I apologize for all of the confusion, 100% my fault.

**Comment by splisson** [comment]
The existing trace for sure install did not have it before 

Sébastien

**Comment by thomastaylor312** [comment]
Yeah, in `install` you shouldn't need the `if settings.Debug` block, just the call to `printRelease` should be switched to `i.printRelease`

**Comment by thomastaylor312** [comment]
@bacongobbler Yeah, this has evolved since my original comment when I mentioned you 🙂

**Comment by bacongobbler** [comment]
ping

**Comment by splisson** [comment]
I added the change in install.go , Rebased and resolved conflicts. The test that is failing now is looking for a file I don't see in master on main repository neither

**Comment by splisson** [comment]
Looks ok now

**Comment by bacongobbler** [comment]
hmm, looks like the rebase went bad actually. Could you please fix this PR? Thanks so much!

**Review by bacongobbler — CHANGES_REQUESTED** [review]
needs a clean rebase

**Comment by splisson** [comment]
the rebase included 200+ previous commits that happened between the time I did my last commit and now. Isn't that expected ?

**Comment by splisson** [comment]
I'll prepare a fresh PR from a new branch

**Comment by splisson** [comment]
Closing. Replaced by PR https://github.com/kubernetes/helm/pull/4207

**Comment by splisson** [comment]
@bacongobbler I submitted a new PR #4207

**Pull request closed without merging**